doc/source/ray-air/examples/opt_deepspeed_inference.ipynb

```diff
@@ -0,0 +1,913 @@
+       "            <tr>\n",
+       "    <td style=\"text-align: left\"><b>Dashboard:</b></td>\n",
+       "    <td style=\"text-align: left\"><b><a href=\"http://console.anyscale.com/api/v2/sessions/ses_jgkdnu2723aleytwqqhebr12vs/services?redirect_to=dashboard\" target=\"_blank\">http://console.anyscale.com/api/v2/sessions/ses_jgkdnu2723aleytwqqhebr12vs/services?redirect_to=dashboard</a></b></td>\n",
+       "</tr>\n",
+       "\n",
+       "        </table>\n",
+       "    </div>\n",
+       "</div>\n"
+      ],
+      "text/plain": [
+       "RayContext(dashboard_url='console.anyscale.com/api/v2/sessions/ses_jgkdnu2723aleytwqqhebr12vs/services?redirect_to=dashboard', python_version='3.9.15', ray_version='3.0.0.dev0', ray_commit='7474e9771fefb1393377d5517843dd03717aec7d', address_info={'node_ip_address': '172.31.175.114', 'raylet_ip_address': '172.31.175.114', 'redis_address': None, 'object_store_address': '/tmp/ray/session_2023-04-13_02-06-58_322723_137/sockets/plasma_store', 'raylet_socket_name': '/tmp/ray/session_2023-04-13_02-06-58_322723_137/sockets/raylet', 'webui_url': 'console.anyscale.com/api/v2/sessions/ses_jgkdnu2723aleytwqqhebr12vs/services?redirect_to=dashboard', 'session_dir': '/tmp/ray/session_2023-04-13_02-06-58_322723_137', 'metrics_export_port': 55537, 'gcs_address': '172.31.175.114:9031', 'address': '172.31.175.114:9031', 'dashboard_agent_listen_port': 52365, 'node_id': '9bad912b21ec0245167c741d3ad124a82dd11dfc767320ad8daf045f'})"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import ray\n",
+    "\n",
+    "ray.init(\n",
+    "    address=\"auto\",\n",
+    "    runtime_env={\n",
+    "        \"pip\": [\n",
+    "            \"numpy==1.23\",\n",
+    "            \"protobuf==3.20.0\",\n",
+    "            \"transformers==4.27.2\",\n",
+    "            \"accelerate==0.17.1\",\n",
+    "            \"deepspeed==0.8.3\",\n",
+    "        ],\n",
+    "        \"env_vars\": {\n",
+    "            \"HF_HUB_DISABLE_PROGRESS_BARS\": \"1\",\n",
+    "        }\n",
+    "    }\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "613df744",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Configurations and Hyperparameters\n",
+    "\n",
+    "from dataclasses import dataclass\n",
+    "\n",
+    "\n",
+    "# Following configuration tested for a single p3.16xlarge instance with 16 V100 GPUs.\n",
+    "\n",
+    "@dataclass\n",
+    "class Config:\n",
+    "    \"\"\"See https://deepspeed.readthedocs.io/en/stable/inference-init.html#deepspeed.inference.config.DeepSpeedInferenceConfig\n",
+    "    for more details about the configurations.\n",
+    "    \"\"\"\n",
+    "    model_name: str = \"facebook/opt-30b\"\n",
+    "    # Path to HuggingFace cache directory. Default is ~/.cache/huggingface/.\n",
+    "    cache_dir: str = None\n",
```

Comment:
```suggestion
    "    cache_dir: Optional[str] = None\n",
```
(and in other args that can take in None)

Reply:
done, good call.